xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -907,9 +917,13 @@ void CXBMCApp::UpdateSessionState()
   else
     state = CJNIPlaybackState::STATE_STOPPED;
 
-  builder.setState(state, pos, speed, CJNISystemClock::elapsedRealtime())
-      .setActions(CJNIPlaybackState::PLAYBACK_POSITION_UNKNOWN);
-  m_mediaSession->updatePlaybackState(builder.build());
+  if ((oldPlayState != m_playback_state) || !m_mediaSessionUpdated)
+  {
+    builder.setState(state, pos, speed, CJNISystemClock::elapsedRealtime())
+        .setActions(CJNIPlaybackState::PLAYBACK_POSITION_UNKNOWN);
```

Comment:
I've not checked if it affects and could be corrected later, `setActions()` should receive as a parameter the current actions available in this session, not a position value.

https://developer.android.com/reference/android/media/session/PlaybackState.Builder#setActions(long)
https://developer.android.com/training/tv/playback/media-session#state

Reply:
Sorry, I didn't see your [previous comment](https://github.com/xbmc/xbmc/pull/24986#issuecomment-2053701102)

Reply:
I would keep the MASK for now as it is - 0xFFFFFFFFFF would enable them all, which is not correct - but without libjniandroid changes I cannot change that. I would do that afterwards to keep this one out of the other one.

I would need your help here:

> 04-13 17:32:13.214 21634 21634 E AvrcpMediaPlayerWrapper: The callback playback state doesn't match the current state
> 04-13 17:56:28.993 21634 21634 W AvrcpMediaPlayerList: mediaUpdatedCallback(): Tried to update with null state

Especially the "Error" is something youtube does not produce and due to my limiting setup capabilities I don't have an idea how to debug that.

Reply:
Corresponding libjniandroid change: https://github.com/xbmc/libandroidjni/pull/49 - would be great if could give your okay as well.

Reply:
libjniandroid change is ok, all these constants are available at API level 24. 

I'll try to find out what is causing this error message

Reply:
>  E AvrcpMediaPlayerWrapper: The callback playback state doesn't match the current state

In theory this is not an error, when `builder.setState(...)` is called, if more than 500 ms have passed, a different state is considered.

Look at these two methods:
https://android.googlesource.com/platform/packages/apps/Bluetooth/+/master/src/com/android/bluetooth/audio_util/MediaPlayerWrapper.java#468
https://android.googlesource.com/platform/packages/apps/Bluetooth/+/master/src/com/android/bluetooth/audio_util/MediaPlayerWrapper.java#529

> W AvrcpMediaPlayerList: mediaUpdatedCallback(): Tried to update with null state

I don't see this message in the logcat

Reply:
Okay, then we keep it as is, explains at least why I did not find anything obviously wrong in the implementation concerning this part.

The null state I saw on my FireTV, while we definitely always send a proper state - so then let's ignore this for now.